libbeat/metric/system/cgroup/util.go

```diff
@@ -242,7 +245,23 @@ func (r Reader) ProcessCgroupPaths(pid int) (map[string]ControllerPath, error) {
 		// cgroup V2
 		// cgroup v2 controllers will always start with this string
 		if strings.Contains(line, "0::/") {
+			// if you're running inside a container
+			// that's operating with a hybrid cgroups config,
+			// the containerized process won't see the V2 mount
+			// inside /proc/self/mountinfo if docker is using cgroups V1
+			// For this very annoying edge case, revert to the hostfs flag
+			// If it's not set, warn the user that they've hit this.
 			controllerPath := filepath.Join(r.cgroupMountpoints.V2Loc, path)
+			if r.cgroupMountpoints.V2Loc == "" && paths.Paths.Hostfs == "" {
+				logp.L().Errorf(`PID %d contains a cgroups V2 path (%s) but no V2 mountpoint was found.
+This may be because metricbeat is running inside a container on a hybrid system.
+To monitor cgroups V2 processess in this way, mount the unified (V2) hierarchy inside
+the container as /sys/fs/cgroup/unified and start metricbeat with --system.hostfs.`, pid, line)
```

Comment:
Should something about this be added to the docs too?

Reply:
Yep, good point. I'll add a blurb.